src/qibo/quantum_info/metrics.py

```diff
@@ -28,11 +28,126 @@ def purity(state):
     else:
         pur = np.real(np.trace(np.dot(state, state)))
 
+    # this is necessary to remove the float from inside
+    # a 0-dim ndarray
     pur = float(pur)
 
     return pur
 
 
+def concurrence(state, bipartition, validate: bool = True, backend=None):
```

Comment:
Why not call `validate` -> `check_purity`?

Reply:
There's not reason not to